core/src/main/kotlin/com/uber/sdk2/core/ui/UberButton.kt

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright (C) 2024. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.uber.sdk2.core.ui
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.Context
+import android.content.ContextWrapper
+import android.graphics.Color
+import android.graphics.Typeface
+import android.util.AttributeSet
+import android.view.Gravity
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import androidx.annotation.StyleRes
+import androidx.annotation.VisibleForTesting
+import androidx.appcompat.widget.AppCompatButton
+import com.uber.sdk2.core.R
+
+/** [android.widget.Button] that can be used as a button and provides default Uber styling. */
+open class UberButton
+@JvmOverloads
+constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = 0,
+  defStyleRes: Int = R.style.UberButton,
+) : AppCompatButton(context, attrs, defStyleAttr) {
+  @DrawableRes private var backgroundResource = 0
+
+  /**
+   * Constructor.
+   *
+   * @param context the context creating the view.
+   * @param attrs attributes for the view.
+   * @param defStyleAttr the default attribute to use for a style if none is specified.
+   * @param defStyleRes the default style, used only if defStyleAttr is 0 or can not be found in the
+   *   theme.
+   */
+  /**
+   * Constructor.
+   *
+   * @param context the context creating the view.
+   * @param attrs attributes for the view.
+   * @param defStyleAttr the default attribute to use for a style if none is specified.
+   */
+  /**
+   * Constructor.
+   *
+   * @param context the context creating the view.
+   * @param attrs attributes for the view.
+   */
+  /**
+   * Constructor.
+   *
+   * @param context the context creating the view.
+   */
+  init {
+    val uberStyle =
+      UberStyle.getStyleFromAttribute(
+        context,
+        attrs,
+        defStyleRes,
+        R.styleable.UberButton,
+        R.styleable.UberButton_ub__style,
+      )
+    init(context, 0, attrs, defStyleAttr, uberStyle)
```

Comment:
Nit : would prefer to do attribute resolution and defined style in the init block.